examples/platform/silabs/BaseApplication.cpp

```diff
@@ -802,6 +802,10 @@ void BaseApplication::ScheduleFactoryReset()
         {
             Provision::Manager::GetInstance().SetProvisionRequired(true);
         }
+#if SL_WIFI
+        // Removing the matter services on factory reset
+        chip::Dnssd::ServiceAdvertiser::Instance().RemoveServices();
```

Comment:
We clear our thread credentials inside of `ConfigurationManagerImpl::DoFactoryReset()`, there is even a wifi ifdefed section in there, it would make sense to put the wifi clear at the same place.

You could also create a wifi helper equivalent to  `ConfigurationManagerImpl::ClearThreadStack()` for wifi so that all the wifi factory reset is handled at the same place.

Reply:
After discussion, it makes sense to keep this here as our Dnssd is entirely handled by the app server.